discovery-provider/src/tasks/index.py

```diff
@@ -125,12 +126,12 @@ def initialize_blocks_table_if_necessary(db: SessionManager):
     return target_blockhash
 
 
-def get_latest_block(db: SessionManager):
-    latest_block = None
+def get_latest_block(db: SessionManager, retry: bool = False):
     block_processing_window = int(
         update_task.shared_config["discprov"]["block_processing_window"]
     )
     with db.scoped_session() as session:
+        latest_block = None
```

Comment:
moved this inside the scope of with. previously the return was happening outside of the with, so i moved that inside the with too. don't think this should have affected something but seemed strange to declare the variable outside the with, set it in the with, return outside the with